diff --git a/tests/testbase.h b/tests/testbase.h
index c9a3059..4433f1a 100644
--- a/tests/testbase.h
+++ b/tests/testbase.h
@@ -124,14 +124,23 @@ class TestBase::NgfdMock : public QObject
  *    It connects the signals to QEventLoop::quit(). When used together with QDBusInterface, it is
  *    usually too late to connect -- that is why the signalEmitted() signal exists.
  */
-class TestBase::SignalSpy : public QObject, public QSignalSpy
+class TestBase::SignalSpy
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
+   : public QObject, public QSignalSpy
+#else
+   : public QSignalSpy
+#endif
 {
     Q_OBJECT
 
 public:
     SignalSpy(QObject *object, const char *signal)
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
         : QObject()
         , QSignalSpy(object, signal)
+#else
+        : QSignalSpy(object, signal)
+#endif
     {
         connect(object, signal, this, SIGNAL(signalEmitted()));
     }
